README.md

```diff
@@ -98,5 +98,17 @@ put the Flutter SDK in `/Applications/flutter`:
   (flutter-sdk-path "/Applications/flutter/"))
 ```
 
+## .dir-locals.el
+
+`flutter-args` is used to set default CLI flags passed to `flutter`.
+This variable can be set per project using a `.dir-locals.el` file in project root.
+The following `.dir-locals.el` sets `flutter-args` to build linux executable and
+enables hot reloading on file save using `after-save-hook`.
+
+``` elisp
+((dart-mode . ((flutter-args . "-d linux")
+               (eval . (add-hook 'after-save-hook 'flutter-run-or-hot-reload nil t)))))
+```
+
```

Comment:
I'm not sure dir-locals deserves an entire section. I would document `flutter-run-args` alongside the other configuration options above, and in the description suggest that it can be set in dir-locals.

My feeling as a medium-term Emacs user is that supplying a `.dir-locals.el` snippet is unnecessary, as you can get Emacs to generate it for you with `M-x add-dir-local-variable`.

As for the after-save-hook idea: It's a good thing to suggest, but it's not specific to dir-locals. I would suggest it at the bottom of the configuration section, without mention of dir-locals.

Reply:
I have removed this entirely. If `after-save-hook` is set up in config it will run `flutter-run-or-hot-reload` on save in every mode, and that's probably not desired.

Reply:
I think for `after-save-hook` we would want `flutter-maybe-run-or-hot-reload` which would only do something if a flutter process is already running. That's a decent idea, but it's probably a different PR.